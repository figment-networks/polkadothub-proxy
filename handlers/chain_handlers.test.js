const chainHandlers = require('./chain_handlers');
const {createApi} = require('../tests/shared');

// These actually DO hit the API
describe('integration', () => {
  let call = {request: {height: 2003930}};
  let api;

  beforeAll(async (done) => {
    api = await createApi();
    done()
  });

  describe('getHead', () => {
    call = {request: {}}
    it('returns head with appropriate keys', async () => {
      const response = await chainHandlers.getHead(api, call)
      const keys = Object.keys(response).sort();
      expect(keys).toEqual(['height', 'time', 'session', 'era'].sort());
    });
  });

  describe('getStatus', () => {
    it('returns status with appropriate keys', async () => {
      const response = await chainHandlers.getStatus(api, call)
      const keys = Object.keys(response).sort();
      expect(keys).toEqual(['chainName', 'chainType', 'clientInfo', 'genesisHash', 'nodeHealth', 'nodeLocalPeerUid', 'nodeName', 'nodeProperties', 'nodeRoles', 'nodeVersion'].sort());
    });
  });

  describe('getMetaByHeight', () => {
    call = {request: {height: 2003930}};
    it('returns a result that matches snapshot', async () => {
      expect(await chainHandlers.getMetaByHeight(api, call)).toMatchSnapshot();
    });
  });
});
