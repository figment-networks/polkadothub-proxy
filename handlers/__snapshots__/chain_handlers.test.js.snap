// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration getMetaByHeight returns a result that matches snapshot 1`] = `
Object {
  "chain": "Kusama",
  "era": 0,
  "lastInEra": false,
  "lastInSession": false,
  "session": 0,
  "specVersion": "1020",
  "time": Object {
    "nanos": 0,
    "seconds": 0,
  },
}
`;
